src/Features/CSharp/Portable/SplitOrMergeIfStatements/CSharpIfLikeStatementGenerator.cs

```diff
@@ -0,0 +1,151 @@
+
+            while (ifStatement.Else?.Statement is IfStatementSyntax elseIfStatement)
+            {
+                builder.Add(elseIfStatement);
+                ifStatement = elseIfStatement;
+            }
+
+            if (ifStatement.Else != null)
+            {
+                builder.Add(ifStatement.Else);
+            }
+
+            return builder.ToImmutable();
+        }
+
+        public SyntaxNode WithCondition(SyntaxNode ifLikeStatement, SyntaxNode condition)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+            return ifStatement.WithCondition((ExpressionSyntax)condition);
+        }
+
+        public SyntaxNode WithStatement(SyntaxNode ifLikeStatement, SyntaxNode statement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+            return ifStatement.WithStatement(SyntaxFactory.Block((StatementSyntax)statement));
+        }
+
+        public SyntaxNode WithStatementsOf(SyntaxNode ifLikeStatement, SyntaxNode otherIfLikeStatement)
+        {
+            var ifStatement = (IfStatementSyntax)ifLikeStatement;
+            var otherIfStatement = (IfStatementSyntax)otherIfLikeStatement;
+            return ifStatement.WithStatement(otherIfStatement.Statement);
+        }
+
+        public SyntaxNode WithElseLikeClausesOf(SyntaxNode ifStatement, SyntaxNode otherIfStatement)
+        {
+            return ((IfStatementSyntax)ifStatement).WithElse(((IfStatementSyntax)otherIfStatement).Else);
+        }
+
+        public SyntaxNode ToIfStatement(SyntaxNode ifLikeStatement)
+            => ifLikeStatement;
+
+        public SyntaxNode ToElseIfClause(SyntaxNode ifLikeStatement)
+            => ((IfStatementSyntax)ifLikeStatement).WithElse(null);
+
+        public void InsertElseIfClause(SyntaxEditor editor, SyntaxNode afterIfLikeStatement, SyntaxNode elseIfClause)
+        {
+            var ifStatement = (IfStatementSyntax)afterIfLikeStatement;
+            var elseIfStatement = (IfStatementSyntax)elseIfClause;
+
+            var newElseIfStatement = elseIfStatement.WithElse(ifStatement.Else);
+            var newIfStatement = ifStatement.WithElse(SyntaxFactory.ElseClause(newElseIfStatement));
+
+            if (ifStatement.Else == null && ContainsEmbeddedIfStatement(ifStatement))
+            {
+                // If the if statement contains an embedded if statement (not wrapped inside a block), adding an else
+                // clause might introduce a dangling else problem (the 'else' would bind to the inner if statement),
+                // so if there used to be no else clause, we'll insert a new block to prevent that.
+                newIfStatement = newIfStatement.WithStatement(SyntaxFactory.Block(newIfStatement.Statement));
+            }
+
+            editor.ReplaceNode(ifStatement, newIfStatement);
+        }
+
+        public void RemoveElseIfClause(SyntaxEditor editor, SyntaxNode elseIfClause)
```

Comment:
on hte fence about the name.   i think there should be some docs indicating an 'IfLike' structure is an 'If, a sequence of elseifs, and a final else'.  So 'RemoveElseIfClause' is just removing one of those from the sequence in the middle.

Reply:
There are doc comments on the interface.

Reply:
➡️ Expanded the doc comment in 2e1211792d92ac123dcb52e68e97b32abcc69f59